From 29db939b06c4535a3b20486ecbc03b7a99de11ea Mon Sep 17 00:00:00 2001
From: Camelia Groza <camelia.groza@freescale.com>
Date: Wed, 16 Mar 2016 17:38:37 +0200
Subject: [PATCH 1207/1429] fsl_qbman: ceetm: fix group CQ id restrictions

Group A CQs can have ids from the 8 - 15 interval, while group B CQs can
have ids from the 12 - 15 interval.

Signed-off-by: Camelia Groza <camelia.groza@freescale.com>
[Original patch taken from QorIQ-SDK-V2.0-20160527-yocto]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/staging/fsl_qbman/qman_high.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/fsl_qbman/qman_high.c b/drivers/staging/fsl_qbman/qman_high.c
index 74a9046..907dbb6 100644
--- a/drivers/staging/fsl_qbman/qman_high.c
+++ b/drivers/staging/fsl_qbman/qman_high.c
@@ -4664,7 +4664,7 @@ int qman_ceetm_cq_claim_A(struct qm_ceetm_cq **cq,
 	struct qm_ceetm_cq *p;
 	struct qm_mcc_ceetm_cq_config cq_config;
 
-	if ((idx < 7) || (idx > 15)) {
+	if ((idx < 8) || (idx > 15)) {
 		pr_err("This grouped class queue id is out of range\n");
 		return -EINVAL;
 	}
@@ -4712,7 +4712,7 @@ int qman_ceetm_cq_claim_B(struct qm_ceetm_cq **cq,
 	struct qm_ceetm_cq *p;
 	struct qm_mcc_ceetm_cq_config cq_config;
 
-	if ((idx < 11) || (idx > 15)) {
+	if ((idx < 12) || (idx > 15)) {
 		pr_err("This grouped class queue id is out of range\n");
 		return -EINVAL;
 	}
-- 
1.7.5.4

